packages/react/src/utils/styles.ts

```diff
@@ -0,0 +1,50 @@
+import type { CSSProperties, ReactNode } from "react";
+
+export type StyleField<CallbackArg> =
+  | string
+  | CSSProperties
+  | ((arg: CallbackArg) => string | CSSProperties);
+export type ContentField<CallbackArg> =
+  | ReactNode
+  | ((arg: CallbackArg) => ReactNode);
+
+export const styleFieldToClassName = <T>(
+  styleField: StyleField<T> | undefined,
+  args: T,
+) => {
+  if (typeof styleField === "string") return styleField;
+  if (typeof styleField === "function") {
+    const result = styleField(args);
+
+    if (typeof result === "string") return result;
```

Comment:
may want to error/warn if we don't get a string at this point?

Reply:
Considering current logic it will always create warnings. Let me explain. 

The functions `styleFieldToClassName` and `styleFieldToCssObject` are used to extract proper value in proper places
Properties of `appearance` prop accept either string, object or callback that may return either string or object. 

To properly handle it, I created this functions to either extract wanted value of return null
![image](https://github.com/pingdotgg/uploadthing/assets/55754540/b1590ee8-0a85-441d-93e7-8844f89ac7aa)

Hope it makes sense for you